ci/redmine_bugzilla.py

```diff
@@ -0,0 +1,153 @@
+EMAIL_RECEIVERS constant specifies the recipient list.
+
+2) For all Redmine issues that track Bugzilla bugs, ensure that the
+Bugzilla external tracker description, status, and priority all mirror the
+Redmine issue state.
+
+Requirements:
+The Redmine and Bugzilla credentials are expected from to be read from a
+file in ini format. See the example below. The location of this file is set
+by the REDMINE_BUGZILLA_CONF environment variable.
+
+<SNIP>
+[bugzilla]
+username = XXXX
+password = XXXX
+
+[redmine]
+key = XXXX
+</SNIP>
+"""
+
+
+import ConfigParser
+import os
+
+from bugzilla.rhbugzilla import RHBugzilla
+from redmine import Redmine
+import urllib3.contrib.pyopenssl
+
+# Here due to InsecurePlatformWarning
+# https://urllib3.readthedocs.org/en/latest/security.html#insecureplatformwarning
+urllib3.contrib.pyopenssl.inject_into_urllib3()
+
+
+BUGZILLA_URL = 'https://bugzilla.redhat.com'
+REDMINE_URL = 'https://pulp.plan.io'
+
+
+EMAIL_RECEIVERS = ['bmbouter@gmail.com']
+FROM_ADDRESS = 'bbouters@redhat.com'
+EMAIL_SUBJECT = 'Pulp upstream/downstream Bug Report'
+
+
+def get_bugzilla_connection(user, password):
+    """
+    Return the Bugzilla connection.
+
+    :param user: The username to connect with
+    :type user: basestring
+    :param password: The password to connect with
+    :type password: basestring
+
+    :return: An instantiated Bugzilla connection object.
+    """
+    return RHBugzilla(url='%s/xmlrpc.cgi' % BUGZILLA_URL, user=user, password=password)
+
+
+def get_redmine_connection(key):
+    """
+    Return the Redmine connection.
+
+    :param key: The api key to connect with
+    :type key: basestring
+
+    :return: An instantiated Redmine connection object.
+    """
+    return Redmine(REDMINE_URL, key=key)
+
+
+def main():
+    config = ConfigParser.ConfigParser()
+    config.read(os.environ['REDMINE_BUGZILLA_CONF'])
+
+    username = config.get('bugzilla', 'username')
+    password = config.get('bugzilla', 'password')
+    key = config.get('redmine', 'key')
+
+    redmine = get_redmine_connection(key)
+    BZ = get_bugzilla_connection(username, password)
+
+    redmine_issues = [issue for issue in redmine.issue.filter(query_id=24)]
+
+    non_closed_bug_with_ext_tracker = BUGZILLA_URL + '/buglist.cgi?bug_status=NEW&bug_status=ASSIGNED&bug_status=POST&bug_status=MODIFIED&bug_status=ON_DEV&bug_status=ON_QA&bug_status=VERIFIED&bug_status=RELEASE_PENDING&columnlist=priority%2Cbug_severity%2Cbug_status%2Cshort_desc%2Cchangeddate%2Ccomponent%2Ctarget_release%2Cassigned_to%2Creporter&f1=external_bugzilla.url&list_id=3309842&o1=substring&query_format=advanced&v1=pulp.plan.io'
+    bugzilla_bugs = BZ.query(RHBugzilla.url_to_query(non_closed_bug_with_ext_tracker))
+
+    links_issues_record = ''
+    ext_bug_record = ''
+
+    for issue in redmine_issues:
+        for custom_field in issue.custom_fields.resources:
+            if custom_field['name'] == 'Bugzilla':
```

Comment:
I don't think so. It's an unordered list of dictionaries and I have to iterate to find the one whose index has the right data in its dict.